attachment_queue/models/attachment_queue.py

```diff
@@ -0,0 +1,133 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+
+from odoo.addons.queue_job.exception import RetryableJobError
+
+_logger = logging.getLogger(__name__)
+
+DEFAULT_ETA_FOR_RETRY = 60 * 60
+STR_ERR_ATTACHMENT_RUNNING = (
+    "The attachment is currently flagged as being in processing"
+)
+STR_ERROR_DURING_PROCESSING = "Error during processing of attachment_queue id {}: \n"
+
+
+class AttachmentQueue(models.Model):
+    _name = "attachment.queue"
+    _description = "Attachment Queue"
+    _inherits = {"ir.attachment": "attachment_id"}
+    _inherit = ["mail.thread"]
+
+    attachment_id = fields.Many2one(
+        "ir.attachment",
+        required=True,
+        ondelete="cascade",
+        help="Link to ir.attachment model ",
+    )
+    file_type = fields.Selection(
+        selection=[],
+        help="The file type determines an import method to be used "
+        "to parse and transform data before their import in ERP or an export",
+    )
+    date_done = fields.Datetime()
+    state = fields.Selection(
+        [("pending", "Pending"), ("failed", "Failed"), ("done", "Done")],
+        readonly=False,
+        required=True,
+        default="pending",
+    )
+    state_message = fields.Text()
+    failure_emails = fields.Char(
+        compute="_compute_failure_emails",
+        help="Comma-separated list of email addresses to be notified in case of"
+        "failure",
+    )
+    running_lock = fields.Boolean()
+
+    @property
+    def _eta_for_retry(self):
+        return DEFAULT_ETA_FOR_RETRY
+
+    @property
+    def _job_attrs(self):
+        return {"channel": "Attachment queues"}
+
+    def _schedule_jobs(self):
+        for el in self:
+            el.with_delay(**self._job_attrs).run()
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super().create(vals_list)
+        res._schedule_jobs()
+        return res
+
+    def button_reschedule(self):
+        self.state = "pending"
+        self.state_message = ""
+        self._schedule_jobs()
+
+    def button_manual_run(self):
+        if self.running_lock:
+            raise UserError(STR_ERR_ATTACHMENT_RUNNING)
+        self.run()
+
+    def _compute_failure_emails(self):
+        for attach in self:
+            attach.failure_emails = attach._get_failure_emails()
+
+    def _get_failure_emails(self):
+        # to be overriden in submodules implementing the file_type
+        self.ensure_one()
+        return ""
+
+    def run(self):
+        """
+        Run the process for an individual attachment queue
+        """
+        if self.state != "pending":
+            return
+        if self.running_lock is True:
```

Comment:
I am not sure about this running_lock. It is set to True, but only for this transaction, it can't be read as True for a parallel transaction, since it is not commited.

What happens here, if the job run and then it is manually lauched, the job write this `running_lock` and go ahead.
The manual run then try to change the `running_lock` and fail due to a conccurent update error and then retry.
If the job is finished, then it will do nothing as the state won't be pending anymore.
So... It actually work but it is not perfect I guess because would work the same without this `running_lock` I guess (the concurrent update would be on writing the state for instance).
And the issue is that the user will have to wait for all this to happen...

Here is another proposal : 
3 methods : 
1) Run manually : can be done if state is pending or failed. No need to catch error, since it is synchronous, we raise the real error directly to the user and he can do what he has to do.
```
def run_manual(self):
try:
    self._cr.execute(f"""                                                           
        SELECT id
        FROM attachment_queue
        WHERE id  = %s
        FOR UPDATE NOWAIT""", self.id)
except psycopg2.OperationalError:
    raise UserError(the task is already running in a job)
if self.state != 'done'
    self.run()
```

2) run as a hob : the job should not fail and if the attachment processing fails, we put it to fail state and write the error (and eventually send the failure email)
```
def run_as_job(self):

try:
    self._cr.execute(f"""                                                           
        SELECT id
        FROM attachment_queue
        WHERE id  = %s
        FOR UPDATE NOWAIT""", self.id)
except psycopg2.OperationalError:
    raise RetryableJobError('Already runing, try again later', seconds=60, ignore_retry=True)
if self.state != 'pending':
  try:
      with savepoint():
          self.run()
      except Exception:
          self.write({"state": "failed", "state_message": str(e))
       emails = self.failure_emails
      if emails:
          send_mail()
```

3) common method to manual and job call, just run the attachment normally and write the state to done if no error was raised
```
def run(self)
  self._run()
  self.write({'state': 'done', 'date_done': now})
```
What do you think ?